### PR TITLE
Add middleware redirection in vite

### PR DESCRIPTION
**Pull request by ildyria**
support for multiple `/` end-points so that F5 no longer fails when doing local dev with npm only (using `VITE_LOCAL_DEV` and `VITE_HTTP_PROXY_TARGET`)

### CODE REVIEW

**Review by tinohager — APPROVED** [review]

File: vite.config.ts [hunk 1]
```diff
@@ -1,108 +1,142 @@
...
-                if (id.includes('node_modules')) {
-                    return id.toString().split('node_modules/')[1].split('/')[0].toString();
-                }
-            }
-        }
-    }
+const localDevelopMiddleware: PluginOption = {
+  name: "develop-rewrite-middleware",
+  configureServer(server) {
+    const viteIndexPath = "/vite/index.html";
```
Comment (d7415) [inline vite.config.ts:15]: I guess if it's local we don't need subpath handling.

File: vite.config.ts [hunk 2]
```diff
@@ -1,108 +1,142 @@
...
+      }
+
+      const requestUrl : string = req.url;
+      const doNotTouch = ["/@", "/api", "/resources", "/node_modules"];
+      if (doNotTouch.some((page) => requestUrl.startsWith(page))) {
+        next();
+        return;
+      }
+
+      console.log(req.url);
```
Comment (d7415) [inline vite.config.ts:30]: Do we want the debug code?

File: vite.config.ts [hunk 3]
```diff
@@ -1,108 +1,142 @@
...
-    }
+		if (env.VITE_LOCAL_DEV === "true") {
+			if (env.VITE_HTTP_PROXY_TARGET === undefined) {
+				throw new Error("VITE_HTTP_PROXY_TARGET is missing");
+			}
+			config.server.open = "vite/index.html";
+			config.server.proxy = {
+				"/api/": env.VITE_HTTP_PROXY_TARGET,
+			};
+      config.plugins.push(localDevelopMiddleware);
```
Comment (d7415) [inline vite.config.ts]:
```suggestion
			config.plugins.push(localDevelopMiddleware);
```

**Review by d7415 — APPROVED** [review]

File: vite.config.ts (hunk 1 above)
Comment (ildyria) [reply vite.config.ts:15]: yes.
in this specific case when you do `npm run dev` it is poping an server and redirecting you to `http://localhost:5173/vite/index.html` which we use to load the app.ts

File: vite.config.ts (hunk 2 above)
Comment (ildyria) [reply vite.config.ts:30]: better safe on that one, it will only be visible if using local dev without the composer part.

**Pull request merged**